src/views/News/index.js

```diff
@@ -40,537 +44,319 @@ const styles = {
-      this.state.newPostSubject === '' ||
-      this.state.newPostBody === '';
-    let content;
-
-    /* Used to re-render the page when the network connection changes.
-     *  this.state.network is compared to the message received to prevent
-     *  multiple re-renders that creates extreme performance lost.
-     *  The origin of the message is checked to prevent cross-site scripting attacks
-     */
-    window.addEventListener('message', (event) => {
-      if (
-        event.data === 'online' &&
-        this.state.network === 'offline' &&
-        event.origin === window.location.origin
-      ) {
-        this.setState({ network: 'online' });
-      } else if (
-        event.data === 'offline' &&
-        this.state.network === 'online' &&
-        event.origin === window.location.origin
-      ) {
-        this.setState({ network: 'offline' });
-      }
-    });
-
-    /* Gets status of current network connection for online/offline rendering
-     *  Defaults to online in case of PWA not being possible
-     */
-    const networkStatus = JSON.parse(localStorage.getItem('network-status')) || 'online';
-
-    if (this.props.authentication) {
-      if (this.state.loading === true) {
-        content = <GordonLoader />;
-      } else if (this.state.news.length > 0 || this.state.personalUnapprovedNews.length > 0) {
-        content = (
-          <NewsList
-            news={this.state.filteredNews}
-            personalUnapprovedNews={this.state.personalUnapprovedNews}
-            updateSnackbar={this.updateSnackbar}
-            currentUsername={this.state.currentUsername}
-            callFunction={this.callFunction}
-          />
-        );
-      } else {
-        content = (
-          <Grid item>
-            <Typography variant="h4">No News To Show</Typography>
-          </Grid>
-        );
-      }
-
-      let submitButton = (
-        <Button
-          variant="contained"
-          color="primary"
-          onClick={this.handleSubmit.bind(this)}
-          disabled={submitButtonDisabled}
-        >
-          Submit
-        </Button>
-      );
-      let editButton = (
-        <Button
-          variant="contained"
-          color="primary"
-          onClick={this.handleUpdate.bind(this)}
-          disabled={submitButtonDisabled}
-        >
-          Update
-        </Button>
-      );
-
-      let news;
-      // If the user is online
-      if (
-        networkStatus === 'online' ||
-        (networkStatus === 'offline' && this.props.authentication)
-      ) {
-        news = (
-          <section>
-            {/* Button to Create Posting */}
-            <Fab
-              variant="extended"
-              color="primary"
-              onClick={this.handlePostClick.bind(this)}
-              style={styles.fab}
-            >
-              <PostAddIcon />
-              Post Listing
-            </Fab>
-
-            <Grid container justify="center">
-              {/* Search */}
```

Comment:
If you look at the current Develop code (which can be seen on the 360 and 360train sites), search is not working in that code. I simply removed it until it can be fixed. See my other comment for where it is broken. It probably won't take much to fix, but that wasn't in the purview of this PR.

Reply:
You're correct. I thought search wasn't working in develop, but it is. I think this is still in need of some fixing and currently no filtering is actually enabled but that's a separate feature. I will restore search to parity with develop and leave the full implementation to a separate pr.